cloudstack/loadbalancer.go

```diff
@@ -290,17 +285,18 @@ func (cs *CSCloud) EnsureLoadBalancerDeleted(ctx context.Context, clusterName st
 		return err
 	}
 
-	if !lb.cloud.environments[lb.cloud.environment].removeLBs {
-		klog.V(3).Infof("skipping deletion of load balancer %s: environment has removals disabled.", lb.ip)
+	if !isLBRemovalEnabled(lb, service) {
+		klog.V(3).Infof("skipping deletion of load balancer %s: service or environment has removals disabled.", lb.ip)
 		return nil
 	}
 
-	manage, err := shouldManageLB(lb, service)
-	if err != nil {
-		return fmt.Errorf("failed to check if LB should be managed: %v", err)
+	if !shouldManageLB(lb, service) {
+		klog.V(3).Infof("Skipping EnsureLoadBalancerDeleted for service %s/%s and LB %q", service.Namespace, service.Name, lb.ip)
+		return nil
 	}
-	if !manage {
-		klog.V(3).Infof("Skipping EnsureLoadBalancerDeleted for service %s/%s and LB %s", service.Namespace, service.Name, lb.ip)
+
+	if lb.rule == nil {
```

Comment:
I think this condition must be evaluated before `shouldManageLB` call as it assumes `lb.rule != nil`.

Reply:
Indeed.